warehouse/query-core/src/test/java/datawave/query/jexl/functions/GroupingFiltersIngest.java

```diff
@@ -0,0 +1,437 @@
+package datawave.query.jexl.functions;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.client.BatchWriter;
+import org.apache.accumulo.core.client.BatchWriterConfig;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.user.SummingCombiner;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.hadoop.io.Text;
+
+import datawave.data.ColumnFamilyConstants;
+import datawave.data.hash.UID;
+import datawave.data.type.LcNoDiacriticsType;
+import datawave.data.type.NumberType;
+import datawave.data.type.Type;
+import datawave.ingest.protobuf.Uid;
+import datawave.query.QueryTestTableHelper;
+import datawave.util.TableName;
+
+public class GroupingFiltersIngest {
+
+    public enum Range {
+        SHARD, DOCUMENT
+    }
+
+    private static final Type<?> lcNoDiacriticsType = new LcNoDiacriticsType();
+    private static final Type<?> numberType = new NumberType();
+    private static final String datatype = "test";
+    private static final String date = "20130101";
+    private static final String shard = date + "_0";
+    private static final ColumnVisibility columnVisibilityEnglish = new ColumnVisibility("ALL&E");
+    private static final ColumnVisibility columnVisibilityItalian = new ColumnVisibility("ALL&I");
+    private static final Value emptyValue = new Value(new byte[0]);
+    private static final long timeStamp = 1356998400000L;
+
+    public static final String corleoneUID = UID.builder().newId("Corleone".getBytes(), (Date) null).toString();
+    public static final String sopranoUID = UID.builder().newId("Soprano".getBytes(), (Date) null).toString();
+    public static final String caponeUID = UID.builder().newId("Capone".getBytes(), (Date) null).toString();
+
+    public static void writeItAll(AccumuloClient client, String range) throws Exception {
+        writeItAll(client, Range.valueOf(range));
+    }
+
+    public static void writeItAll(AccumuloClient client, Range range) throws Exception {
+
+        BatchWriter bw = null;
+        BatchWriterConfig bwConfig = new BatchWriterConfig().setMaxMemory(1000L).setMaxLatency(1, TimeUnit.SECONDS).setMaxWriteThreads(1);
+        Mutation mutation;
+
+        try {
+            // write the shard table :
+            bw = client.createBatchWriter(TableName.SHARD, bwConfig);
+            mutation = new Mutation(shard);
+
+            mutation.put(datatype + "\u0000" + corleoneUID, "NAME.BAT.ZIP.FOO.0" + "\u0000" + "SANTINO", columnVisibilityItalian, timeStamp, emptyValue);
```

Comment:
Most all of our data fields name have only a group, subgroup, and instance.  So that would be something like NAME.BAT.FOO.0 (i.e. only 4-tuples).  I would like to make sure that things are working with data structured like that.  You can have a mix of data if you want to test it with additional grouping.